aws/delete_s3_velero_bucket.py

```diff
@@ -0,0 +1,309 @@
+
+        LOGGER.info(
+            f"Verified cluster '{cluster_name}' is associated with velero infrastructure name {velero_infrastructure_name}",
+        )
+        return True
+
+    return False
+
+
+def delete_all_objects_from_s3_folder(bucket_name: str) -> None:
+
+    """
+    Deletes all files in a folder of an S3 bucket
+
+    Args:
+        bucket_name: The bucket name
+
+    Returns: None
+
+    Raises:
+        DeleteFolderObjectError: if respone status code is not HTTPStatus.OK
+    """
+
+    list_obj_response = S3_CLIENT.list_objects_v2(Bucket=bucket_name)
+
+    #  Sometimes there maybe no contents -- no objects
+    files_in_folder = list_obj_response.get("Contents")
+    if not files_in_folder:
+        LOGGER.info(f"No objects to be deleted for {bucket_name}")
+        return
+
+    files_to_delete = [{"Key": file["Key"]} for file in files_in_folder]
+    delete_response = S3_CLIENT.delete_objects(
+        Bucket=bucket_name,
+        Delete={"Objects": files_to_delete, "Quiet": True},
+    )
+
+    delete_response_http_status_code = delete_response["ResponseMetadata"][
+        "HTTPStatusCode"
+    ]
+    if delete_response_http_status_code == HTTPStatus.OK:
+        LOGGER.info(f"Objects deleted successfully")
+        return
+    else:
+        LOGGER.error(
+            "Objects not deleted:\n:"
+            f"{json.dumps(delete_response, default=str, indent=4)}",
+        )
+
+        raise DeleteFolderObjectError(
+            f'Folder Object Deletion Error: Expecting a {HTTPStatus.OK} Status code:"'
+            f"but got {delete_response_http_status_code}",
+        )
+
+
+def get_cluster_name() -> str:
+    """
+    Get the cluster name passed to the script
+
+    The cluster name can be passed as a command line argument or an environment variable.
+    The script exits if the cluster name is not provided
+
+    Args:
+        None
+
+    Returns:
+        The name of the cluster
+    """
+    parser = argparse.ArgumentParser(description="Delete an s3 velero cluster bucket")
+    parser.add_argument(
+        "-c",
+        "--cluster_name",
+        help=""" Either set the S3_CLUSTER environment variable or
+            Pass the cluster value at the command line when the script is invoked with the -c|--cluster
+        """,
+        default=os.getenv("S3_CLUSTER"),
+    )
+
+    cluster_name = parser.parse_args().cluster_name
+    if cluster_name:
+        return cluster_name
+
+    LOGGER.error(
+        """
+            Required cluster value was not set!
+            Run script with --help for specifics
+        """,
+    )
+    sys.exit(1)
+
+
+def verify_aws_env_requirements_met() -> None:
+
+    """
+    Verify that the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are set.
+    The script exits if neither is set.
+
+    Returns: None
+    """
+    require_env = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
+    for env_var in require_env:
+        if not os.getenv(env_var):
+            LOGGER.error(
```

Comment:
I would collect all the env that not set and log one error with the missing ones